plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -51,10 +69,17 @@ func NewStrategyStore(options Options, logger *zap.Logger) (ss.StrategyStore, er
 
 // GetSamplingStrategy implements StrategyStore#GetSamplingStrategy.
 func (h *strategyStore) GetSamplingStrategy(serviceName string) (*sampling.SamplingStrategyResponse, error) {
-	if strategy, ok := h.serviceStrategies[serviceName]; ok {
-		return strategy, nil
+	h.RLock()
```

Comment:
I think it's sufficient to lock and defer unlock at the beginning of the function. Same for parse function.